File: JS/insertion_sort.js
function insertion_sort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let p = i;
    const temp = arr[i];
    while (p > 0 && arr[p - 1] > temp) {
      arr[p] = arr[p - 1];
      p = p - 1;
    }
    arr[p] = temp;
  }
  return arr;
}

console.log(insertion_sort([65, 55, 2, 111, 45, 69, 35, 25, 15, 10]));

File: go/introductoin-to-algorithms/insertionSort.go
package main

func insertionSort(arr []int) []int {
	arrLen := len(arr)
	for i := 2; i < arrLen; i++ {
		el := arr[i]
		j := i - 1

		for j > 0 && arr[j] > el {
			arr[j+1] = arr[j]
			j = j - 1
		}
		arr[j+1] = el
	}
	return arr
}
